source/extensions/matching/common_inputs/transport_socket/config.h

```diff
@@ -0,0 +1,347 @@
+#pragma once
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/extensions/matching/common_inputs/network/v3/network_inputs.pb.h"
+#include "envoy/extensions/matching/common_inputs/transport_socket/v3/transport_socket_inputs.pb.h"
+#include "envoy/extensions/matching/common_inputs/transport_socket/v3/transport_socket_inputs.pb.validate.h"
+#include "envoy/matcher/matcher.h"
+#include "envoy/network/address.h"
+#include "envoy/network/connection.h"
+#include "envoy/registry/registry.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/common/config/metadata.h"
+
+#include "absl/strings/str_join.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace CommonInputs {
+namespace TransportSocket {
+
+/**
+ * Data structure holding context for transport socket matching.
+ * This provides access to endpoint metadata, locality metadata, and network connection context
+ * that can be used by the matcher framework to select transport sockets.
+ */
+struct TransportSocketMatchingData {
```

Comment:
Why not extend NetworkMatchingData and then reuse the implementation for existing NetworkInput matchers?